### PR TITLE
Add unit tests for Transaction and TransactionFrequency

### PR DESCRIPTION
**Pull request by fterh**

### CODE REVIEW

File: baconTests/model/TransactionFrequencyTests.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    func test_init_recurring_validInput() {
+        let tf = try! TransactionFrequency(nature: .recurring, interval: .daily, repeats: 10)
+        XCTAssertEqual(tf.nature, .recurring)
+        XCTAssertEqual(tf.interval, .daily)
+        XCTAssertEqual(tf.repeats, 10)
+    }
+
+    func test_init_recurring_missingInterval() {
+        XCTAssertThrowsError(try TransactionFrequency(nature: .recurring, interval: nil, repeats: 10))
```
Comment (GilgameshTC) [inline baconTests/model/TransactionFrequencyTests.swift]: Is this repeated?

File: baconTests/model/TransactionFrequencyTests.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    func test_init_recurring_missingRepeats() {
+        XCTAssertThrowsError(try TransactionFrequency(nature: .recurring, interval: .daily, repeats: nil)) { (err) in
+            XCTAssertTrue(type(of: err) == InitializationError.self)
+        }
+    }
+
+    func test_init_recurring_invalidRepeats() {
+        XCTAssertThrowsError(try TransactionFrequency(nature: .recurring, interval: .daily, repeats: -5)) { (err) in
```
Comment (GilgameshTC) [inline baconTests/model/TransactionFrequencyTests.swift:49]: What happens if nature is .recurring but repeats is 0, should it still be treated as a recurring transaction or a oneTime transaction?

File: baconTests/model/TransactionTests.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        XCTAssertEqual(transaction.amount, 1)
+        XCTAssertEqual(transaction.description, "")
+    }
+
+    func test_init_invalidInput() {
+        XCTAssertThrowsError(try Transaction(date: testDate,
+                                             type: .expenditure,
+                                             frequency: testFrequency,
+                                             category: .bills,
+                                             amount: -1)) { (err) in
```
Comment (GilgameshTC) [inline baconTests/model/TransactionTests.swift:35]: what about 0 amount? Is it valid?

Comment (GilgameshTC) [reply baconTests/model/TransactionTests.swift:35]: Should it be an error if it is 0? Then we prompt the user for input, if they attempt to create a transaction with 0 amount.

File: baconTests/model/TransactionFrequencyTests.swift (hunk 2 above)
Comment (fterh) [reply baconTests/model/TransactionFrequencyTests.swift:49]: It should fail (just like -5). I've added a new test case to reflect that.

File: baconTests/model/TransactionTests.swift (hunk 3 above)
Comment (fterh) [reply baconTests/model/TransactionTests.swift:35]: Yup, it should fail (throw an error). I've added a new test case to reflect that.

**Review by kumuwu — APPROVED** [review]

**Pull request merged**